src/stan_math_backend/Stan_math_code_gen.ml

```diff
@@ -775,6 +797,21 @@ let pp_register_map_rect_functors ppf p =
     (list ~sep:cut pp_register_functor)
     (List.sort ~compare (Hashtbl.to_alist map_rect_calls))
 
+let fun_used_in_ode_bdf p =
```

Comment:
this function seems the same as `fun_used_in_reduce_sum` except for the function matching predicate. I'd suggest making a local helper function and pulling out that predicate as an argument